narwhals/_polars/expr.py

```diff
@@ -45,18 +45,18 @@ def func(*args: Any, **kwargs: Any) -> Any:
 
     def cast(self, dtype: DType) -> Self:
         expr = self._native_expr
-        dtype = narwhals_to_native_dtype(dtype, self._dtypes)
-        return self._from_native_expr(expr.cast(dtype))
+        dtype_pl = narwhals_to_native_dtype(dtype, self._dtypes)
+        return self._from_native_expr(expr.cast(dtype_pl))
```

Comment:
we need to rename these else mypy complains that we're assigning an incompatible type to an already-declared variable